numpy/core/tests/test_umath.py

```diff
@@ -701,6 +701,31 @@ def assert_complex_equal(x, y):
                 assert_complex_equal(np.power(zero, -p), cnan)
             assert_complex_equal(np.power(zero, -1+0.2j), cnan)
 
+    # Testing 0^{Non-zero} issue 18378
+    def test_zero_power_nonzero(self):
+        zero = np.array([0.0j])
+        cnan = np.array([complex(np.nan, np.nan)])
+
+        def assert_complex_equal(x, y):
+            assert_array_equal(x.real, y.real)
+            assert_array_equal(x.imag, y.imag)
+
+        #Complex powers with positive real part will not generate a warning
+        assert_complex_equal(np.power(zero, 1+4j), zero)
+        assert_complex_equal(np.power(zero, 2-3j), zero)
+        #Testing zero values when real part is greater than zero
+        assert_complex_equal(np.power(zero, 1+1j), -zero)
+        assert_complex_equal(np.power(zero, 1+0j), -zero)
+        assert_complex_equal(np.power(zero, 1-1j), zero)
+        #Complex powers will negative real part or 0 (provided imaginary 
+        # part is not zero) will generate a NAN and hence a RUNTIME warning
+        with pytest.warns(expected_warning=RuntimeWarning):
```

Comment:
This one should be either repeated every time (alternatively, make it a `pytest.mark.parameterize`).  Of you need to add `as r` or so, and then check the number of warnings.  Otherwise, we only test that at least one of these warned.

Reply:
@seberg I am sorry but I am not able to understand your line ``` Of you need to add as r```. Did you mean in line 772 I add 
```python
with pytest.warns(expected_warning=RuntimeWarning) as r:
    # run all my tests here
#check all warnings by using len(r) == number of tests written above.
```